### PR TITLE
Add nasa extension

### PR DESCRIPTION
**Pull request by mSarheed**
## Description

This Raycast extension allows you to view the Astronomy Picture of the Day, latest news, astronauts, spacecrafts, and upcoming launches from NASA and other space agencies.

## Screencast
nasa-1
nasa-2
nasa-3
nasa-4
nasa-5

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [X] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [X] I checked that files in the `assets` folder are used by the extension itself
- [X] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by xilopaint** [comment]
Cool extension. Here are some suggestions:

> nasa-1

Replace `Nasa.gov` with `nasa.gov` or `NASA website`.


> nasa-2

Remove all emojis. They make the extension look worse and less readable.

> nasa-3

Same as above.

nasa-4 nasa-5

I would remove all the emojis here except the flags and the globe. Have you tried using the flags as the list item's icon instead of the subtitle?

Also, I would use a cool space wallpaper for the metadata images.

File: extensions/nasa/src/astronauts.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  return (
+    <List isShowingDetail>
+      {data.results.map((astronaut) => (
+        <AstronautDetail astronaut={astronaut} key={astronaut.id} />
+      ))}
+    </List>
+  );
+}
```
Comment (pernielsentikaer) [inline extensions/nasa/src/astronauts.tsx:29]:
```suggestion
import { List } from "@raycast/api";
import { useFetch } from "@raycast/utils";
import { ENDPOINTS, HEADERS } from "./constants/prefrences";
import { AstronautsResponse } from "./types/astronauts";
import AstronautDetail from "./componenets/astronaut";

export default function ASTRONAUTS() {
  const { isLoading, data } = useFetch(ENDPOINTS.ASTRONAUTS, {
    headers: HEADERS,
    mapResult(result: AstronautsResponse) {
      return {
        data: result,
      };
    },
  });

  return (
    <List isLoading={isLoading} isShowingDetail>
      {data?.results?.map((astronaut) => <AstronautDetail astronaut={astronaut} key={astronaut.id} />)}
    </List>
  );
}
```
Maybe something like this for the commands instead 🙂

**Comment by mSarheed** [comment]
@pernielsentikaer The extension have been updated with your suggestion.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Two commands crashing for me, could you look into why that might be?

nasa 2024-09-19 at 09 55 48
nasa 2024-09-19 at 09 55 38

Can we handle that gracefully?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- The loading indicator is not stopped when you have a wrong API Key, I saw a failure toast after some time but loader didn't stop

- Should we consider moving the API key down to the only command needing it afaik

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by mSarheed** [comment]
Hi again @pernielsentikaer  
Thanks for the feedback.  
I have updated the extension with the following changes:  
-  Better error handling  
-  The API key is now only needed for the one command that actually requires it.  

Please review again.

**Pull request merged**